src/components/ContentNode.vue

```diff
@@ -100,29 +100,58 @@ function renderNode(createElement, references) {
     ))
   ));
 
-  const renderTableChildren = (rows, headerStyle = TableHeaderStyle.none) => {
+  const renderTableCell = (
+    element, attrs, data, cellIndex, rowIndex, extendedData, cellsToRender,
```

Comment:
Do you have any kind of informal spec that you're working off of? That would be really helpful even for reviewing a draft PR. My initial thoughts is that this seems more complicated than what we previously had discussed, but maybe that's necessary based on the latest expectation of the JSON data for this.

Reply:
I had this - https://github.com/apple/swift-docc/compare/main...QuietMisdreavus:swift-docc:table-span-spec2

This looks like it would send you:

```json
[
  [
   { "rowspan": 1 },
   {}
  ],
  [ 
   {},
   {}
  ]
]
```
where as my idea was:

```json
{
 "0_0": { "rowspan": 1}
}
```

But it felt like it would have to provide data for each cell, and you would still have to create the matrix in order to know which fields to render and which to skip. Unless docc drops extra columns, in which case we talked that would be a breaking change and decided not to do that :/

Reply:
Dobri and i had talked offline about his idea, but i hadn't taken the time to draft it into DocC's RenderNode spec. The idea is that you would create the string key from the x/y coordinate of a cell, and check for it in the `extendedData` property. So borrowing from the test data in the PR, that would look like this:

```js
      const extendedData = {
         '0_0': { colspan: 2 },
         '1_2': { rowspan: 2 },
       };
```

The existing table cell data would still contain all `(rows * columns)` cells in the table, just with empty cells for the ones being spanned over, for backwards compatibility with versions of docc-render without this change.

...i might need help writing this version of `extendedData` into an OpenAPI spec, though 😅

Reply:
Yeah, it may be hard to specify that stringly typed key format in a formal way using OpenAPI. We should make sure to at least document it in the spec though, especially for other renderers that may come along.

Reply:
I've drafted an update to the spec for this format here: https://github.com/apple/swift-docc/compare/main...QuietMisdreavus:swift-docc:table-span-spec3